src/passes/builtinHandler/thisKeyword.ts

```diff
@@ -29,4 +39,93 @@ export class ThisKeyword extends ASTMapper {
       return;
     }
   }
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    if (
+      node.vExpression instanceof MemberAccess &&
+      node.vExpression.vExpression instanceof Identifier &&
+      node.vExpression.vExpression.name === 'this'
+    ) {
+      const replacementCall = createCallToFunction(
```

Comment:
Is there a reason to have this here rather than relying on visitIdentifier to do the transformation when the recursion reaches it?

Reply:
To avoid unnecessary interface creation for the current contract, for eg.
```solidity
contract B{
  address addr = address(this);
}
```
In the above case interface is not needed because there is no any method call to `this`